.github/workflows/build.yml

```
@@ -0,0 +1,25 @@
+# This workflow builds the repo
+
+name: Build
+
+# Triggers the workflow on push or pull request events but only for the main branch
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main, release* ]
+
+jobs:
+  build-ubuntu:
+    runs-on: ubuntu-20.04
+    needs: pre-flight-check
+    steps:
+      - uses: actions/checkout@v3
+      - uses: egor-tensin/setup-gcc@v1
+        with:
+          version: 9
+          platform: x64
+      - uses: jwlawson/actions-setup-cmake@v1.12
+        with:
+          cmake-version: '3.16.3' # version used in Ubuntu 20.04 LTS
+      - run: cmake -B build && cmake --build build
```

After some thinking I would suggest to remove the specific versions of cmake and gcc since it would create a lot of effort to maintain this when it get out of date. I think we are safe to use CMake and gcc provided by the virtual environment: https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2204-Readme.md

We should also point to the latest Ubuntu to avoid the need of updating every time a new ubuntu get's released.

```suggestion
    runs-on: ubuntu-latest
    needs: pre-flight-check
    steps:
      - uses: actions/checkout@v3
      - run: cmake -B build && cmake --build build
```